libsolidity/parsing/Parser.cpp

```diff
@@ -402,12 +408,22 @@ Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(bool _forceEmptyN
 				));
 				m_scanner->next();
 			}
+			else if (_modifierArea && _modifierArea->visibility() != Declaration::Visibility::Default)
+			{
+				parserError("Cannot override modifier area's visibility.");
+				m_scanner->next();
```

Comment:
Can these checks be moved to SyntaxtChecker?

Reply:
Yes